assertion.go

```diff
@@ -82,6 +82,14 @@ const (
 	// [Expected] stores AssertionList with allowed values
 	AssertBelongs
 	AssertNotBelongs
+
+	// Check expression: [Actual] has prefix [Expected]
+	AssertHasPrefix
+	AssertNotHasPrefix
+
+	// Check expression: [Actual] has suffix [Expected]
+	AssertHasSuffix
+	AssertNotHasSuffix
```

Comment:
Usually we add new assertion types when no existing type has formatting suitable for some assertion.

For (Not)HasPrefix/Suffix, I think AssertContainsSubset/AssertNotContainsSubset would fit will. Prefix and suffix can be viewed as a special case of substring checking, with additional requirement where the substring should be located,